Approved/implicitly-scoped-qubit-allocation.md

```diff
@@ -0,0 +1,322 @@
+---
+title: Implicitly-scoped qubit allocation
+description: Provides more convenient syntax for allocating qubits.
+author: Sarah Marshall
+date: October 29, 2020
+---
+
+# Proposal
+
+1. Deprecate `using` and `borrowing` keywords and replace with `use` and `borrow`.
+2. Allow `use` and `borrow` statements to be followed by either a block `{ ... }` or a semicolon `;`.
+3. Deprecate and remove parentheses around block statement headers.
+
+We expect to permanently support the new non-block `use` and `borrow` statements, but we may reconsider how or if the block forms are supported at a later date, after we learn more about how both forms are used in practice.
+
+The parentheses that are currently required by the `for`, `using` or `use`, and `borrowing` or `borrow` statements will be deprecated and removed.
+Including parentheses with these statements will become a syntax error with the next major version.
+The parentheses for the `if`, `elif`, `while`, and `until` statements will no longer be required, but will still be supported because an expression may always be wrapped in parentheses.
+
+# Justification
+
+Qubit allocation is currently awkward in Q# for two reasons:
+
+1. It requires creating a new block to define the lifetime of the qubit, even when the intended lifetime is the same as the current block.
+2. Allocating multiple qubits at the same time requires either tuple destructuring or nested blocks, both of which can be hard to read.
+
+As a quantum programming language, qubit allocation is at the core of Q#.
+Both the simple case of allocating a single qubit for the remainder of the current block, and more complicated scenarios involving multiple qubit variables that are both clean and borrowed and have varying lifetimes, should have clear, intuitive syntax that doesn't require boilerplate.
+
+# Description
+
+Qubits are a managed resource in Q#: when qubits are allocated, they are valid only within the new block declared at allocation.
+They are automatically released no later than the end of the scope in which they were declared, after which they can no longer be used.
+Qubits may also be released earlier than the end of the scope if the compiler can determine that the qubits are no longer used after a particular point.
+
+This proposal adds the option to allocate qubits that are scoped to the current block, instead of a new block.
+Its aim is to be syntactically concise and consistent with other binding statements like `let`.
+As a consequence, it also modifies the existing block qubit allocation syntax slightly in order to increase consistency with the new non-block form.
+
+## Current Status
+
+Currently, the only mechanism to allocate qubits in Q# is with `using` and `borrowing` block statements.
+No new functionality is provided by this proposal.
+Every example with the existing syntax has a corresponding equivalent example with the proposed syntax, and vice versa.
```

Comment:
I'd just remove that sentence, since it is not perfectly correct (in both directions), though I agree with the sentiment.
```suggestion
```